frame/collective/src/lib.rs

```diff
@@ -678,13 +691,15 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			Error::<T, I>::WrongProposalWeight
 		);
 
-		let proposal_hash = T::Hashing::hash_of(&proposal);
-		ensure!(!<ProposalOf<T, I>>::contains_key(proposal_hash), Error::<T, I>::DuplicateProposal);
+		let proposal_bounded = T::Preimages::bound(*proposal.clone())?;
```

Comment:
The problem with this is that `Event::Noted { hash }` is emitted.
It might try to write something in storage that is later removed by a drop:
Is storage smart enough to avoid any write?
If yes, is benchmarking smart enough to avoid it in the computation of the weight?

Reply:
You can maybe bound it manually by just calculating the hash, depending on the length of the proposal. Although that is not guaranteed to work, since there are also inline and legacy bounded values and it could be one of those in the `Voting` map...

Maybe we need to keep the additional hash map for proposals, not sure.